### PR TITLE
[REVIEW] Remove deprecated RMM APIs and headers.

### PR DESCRIPTION
**Pull request by jrhemstad**
Closes #5444

In anticipation of https://github.com/rapidsai/rmm/pull/396, removes all references to deprecated RMM headers and APIs.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/benchmarks/fixture/benchmark_fixture.hpp [hunk 1]
```diff
@@ -51,14 +54,13 @@ namespace cudf {
  * BENCHMARK_REGISTER_F(my_benchmark, my_test_name)->Range(128, 512);
  */
 class benchmark : public ::benchmark::Fixture {
+  std::unique_ptr<rmm::mr::cnmem_memory_resource> mr{
```
Comment (harrism) [inline cpp/benchmarks/fixture/benchmark_fixture.hpp:57]: Why not go straight to using pool_memory_resource here, to accelerate dropping cnmem?

Comment (jrhemstad) [reply cpp/benchmarks/fixture/benchmark_fixture.hpp:57]: Yeah, I had the same thought. Wanted to make sure people were okay with it before I made such a change.

Comment (kkraus14) [reply cpp/benchmarks/fixture/benchmark_fixture.hpp:57]: Lets do this in a follow up PR so we can have further discussion if desired.

**Comment by kkraus14** [comment]
rerun tests

**Pull request merged**